.github/workflows/render.yml

```
@@ -32,8 +36,8 @@ jobs:
 
       - name: 🚀 Deploy to GitHub pages
         # This allows CI to build branches for testing
-        if: github.ref == 'refs/heads/main'
-        uses: JamesIves/github-pages-deploy-action@v4.2.2
+        if: (github.ref == 'refs/heads/main') && (matrix.python-version == '3.x')
+        uses: JamesIves/github-pages-deploy-action@v4.3.4
```

```suggestion
        uses: JamesIves/github-pages-deploy-action@v4.4.0
```

4.4.0 was just released yesterday